meson.build

```
@@ -0,0 +1,533 @@
+pdfium_dep = dependency('pdfium', version: '>=4200', required: get_option('pdfium'))
+if pdfium_dep.found()
+    libvips_deps += pdfium_dep
+    cfg_var.set('HAVE_PDFIUM', '1')
+endif
+
+
+libheif_dep = dependency('libheif', version: '>=0.4.11', required: get_option('heif'))
+if libheif_dep.found()
+    if get_option('heif-module') and modules_enabled
+        cfg_var.set('HEIF_MODULE', '1')
+    else
+        libvips_deps += libheif_dep
+    endif
+    if libheif_dep.get_variable(pkgconfig: 'builtin_h265_decoder', internal: 'builtin_h265_decoder', default_value: 'no') != 'no' or libheif_dep.get_variable(pkgconfig: 'builtin_avif_decoder', internal: 'builtin_avif_decoder', default_value: 'no') != 'no'
+        cfg_var.set('HAVE_HEIF_DECODER', '1')
+    endif
+    if libheif_dep.get_variable(pkgconfig: 'builtin_h265_encoder', internal: 'builtin_h265_encoder', default_value: 'no') != 'no' or libheif_dep.get_variable(pkgconfig: 'builtin_avif_encoder', internal: 'builtin_avif_encoder', default_value: 'no') != 'no'
+        cfg_var.set('HAVE_HEIF_ENCODER', '1')
+    endif
+    if cc.has_function('heif_image_handle_get_raw_color_profile', prefix: '#include <libheif/heif.h>', dependencies: libheif_dep)
+        cfg_var.set('HAVE_HEIF_COLOR_PROFILE', '1')
+    endif
+    if cc.has_member('struct heif_decoding_options', 'convert_hdr_to_8bit', prefix: '#include <libheif/heif.h>', dependencies: libheif_dep)
+        cfg_var.set('HAVE_HEIF_DECODING_OPTIONS_CONVERT_HDR_TO_8BIT', '1')
+    endif
+    # heif_main_brand added in 1.4.0, but heif_avif appeared in 1.7 ... just check
+    # the libheif version number since testing for enums is annoying
+    if libheif_dep.version().version_compare('>=1.7.0')
+        cfg_var.set('HAVE_HEIF_AVIF', '1')
+    endif
+endif
+
+libjxl_dep = dependency('libjxl', version: '>=0.5', required: get_option('jpeg-xl'))
+libjxl_threads_dep = dependency('libjxl_threads', version: '>=0.5', required: get_option('jpeg-xl'))
+if libjxl_dep.found() and libjxl_threads_dep.found()
+    if get_option('jpeg-xl-module') and modules_enabled
+        cfg_var.set('LIBJXL_MODULE', '1')
+    else
+        libvips_deps += libjxl_dep
+        libvips_deps += libjxl_threads_dep
+    endif
+    cfg_var.set('HAVE_LIBJXL', '1')
+    if cc.has_function('JxlEncoderInitBasicInfo', prefix: '#include <jxl/encode.h>', dependencies: libjxl_dep)
+        cfg_var.set('HAVE_LIBJXL_JXLENCODERINITBASICINFO', '1')
+    endif
+endif
+
+libpoppler_dep = dependency('poppler-glib', version: '>=0.16.0', required: get_option('poppler'))
+if libpoppler_dep.found() and pdfium_dep.found()
+    message('PDFium has been found, ignoring Poppler support')
+elif libpoppler_dep.found()
+    if get_option('poppler-module') and modules_enabled
+        cfg_var.set('POPPLER_MODULE', '1')
+    else
+        libvips_deps += libpoppler_dep
+    endif
+    cfg_var.set('HAVE_POPPLER', '1')
+endif
+
+libnifti_dep = cc.find_library('nifti2', has_headers: 'nifti1_io.h', required: false)
```

AFAIK, libvips NIfTI loader/saver has only been tested with the NIfTI-1 library.
```suggestion
libnifti_dep = cc.find_library('niftiio', has_headers: 'nifti1_io.h', required: false)
```